File: in_Java/0883 Projection Area of 3D Shapes.java
class Solution {
    public int projectionArea(int[][] grid) {
        int res = 0;
        int row = grid.length;
        int col = grid[0].length;
        for(int r = 0; r < row; r++){
            int rowmax = 0;
            for(int c = 0; c < col; c++){
                if(rowmax < grid[r][c]){
                    rowmax = grid[r][c];
                }
            }
            res += rowmax;
        }
        for(int c = 0; c < col; c++){
            int colmax = 0;
            for(int r = 0; r < row; r++){
                if(colmax < grid[r][c]){
                    colmax = grid[r][c];
                }
            }
            res += colmax;
        }
        for(int r = 0; r < row; r++){
            for(int c = 0; c < col; c++){
                if(grid[r][c] != 0) res++;
            }
        }
        return res;
    }
}

File: in_Python/0883 Projection Area of 3D Shapes.py
class Solution(object):
    def projectionArea(self, grid):
        """
        :type grid: List[List[int]]
        :rtype: int
        """
        r = len(grid)
        c = len(grid[0])
        res = 0
        rowmax = []
        for i in range(r):
            rowmax.append(-1)
        colmax = []
        for j in range(c):
            colmax.append(-1)
        for i in range(r):
            for j in range(c):
                x = grid[i][j] 
                if x != 0:
                    res += 1
                if x > rowmax[i]:
                    rowmax[i] = x
                if x > colmax[j]:
                    colmax[j] = x
        for i in rowmax:
            res += i
        for i in colmax:
            res += i
        return res
        